networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java

```diff
@@ -193,11 +193,14 @@ private boolean remoteEpochIsInvalid(
         .thenApply(maybeBlock -> blockToSlot(finalizedEpochSlot, maybeBlock))
         .thenCompose(
             blockSlot -> {
+              if (blockSlot.equals(UnsignedLong.valueOf(Constants.GENESIS_SLOT))) {
```

Comment:
I think we can return early before we pull this block by checking if the `finalizedEpochSlot` is the genesis slot

Reply:
We can but we need the existing check as well.  We might have finalised epoch 1 without any blocks ever being produced, so the block in effect would still be the genesis block.